### PR TITLE
policy-engine: add basic tests for graph endpoint

### PR DESCRIPTION
**Pull request by lucab**
This adds initial test infrastructure and some (very basic) negative tests for policy-engine graph endpoint.

/cc @steveeJ

### CODE REVIEW

File: policy-engine/src/config.rs [hunk 1]
```diff
@@ -5,14 +5,17 @@ use hyper::Uri;
...
+
 #[derive(Debug, StructOpt)]
 pub struct Options {
     /// Verbosity level
     #[structopt(short = "v", parse(from_occurrences))]
     pub verbosity: u64,
 
     /// URL for the upstream graph builder or policy engine
-    #[structopt(long = "upstream", default_value = "http://localhost:8080/v1/graph")]
+    #[structopt(long = "upstream", raw(default_value = "DEFAULT_UPSTREAM_URL"))]
```
Comment (steveej) [inline policy-engine/src/config.rs:18]: This is neat. I've been looking for this for quite some time this morning!

File: policy-engine/src/graph.rs [hunk 2]
```diff
@@ -94,3 +94,50 @@ pub(crate) fn index(req: HttpRequest<AppState>) -> Box<Future<Item = HttpRespons
...
+
+#[cfg(test)]
+mod tests {
+    extern crate tokio;
+
+    use self::tokio::runtime::current_thread::Runtime;
+    use crate::graph;
+    use crate::AppState;
+    use actix_web::http;
+    use actix_web::test as atest;
```
Comment (steveej) [inline policy-engine/src/graph.rs]: this doesn't seem worth the obfuscation

**Review by steveej — COMMENTED** [review]
I haven't tested it but it reads fine, with a minor naming nit. Did you want to add more tests or keep the scope as is?

File: policy-engine/src/graph.rs (hunk 2 above)
Comment (lucab) [reply policy-engine/src/graph.rs]: Ack, it was just feeling strange to end up with `test::`. Do you prefer the full form (`actix_web::test::<foo>`) or the short one but without renaming (`test::<foo>`)?

**Comment by lucab** [comment]
@steveeJ for the moment I'd keep the scope like this. I'm still trying to figure out what's the proper way to mock the upstream request in the chain.

File: policy-engine/src/graph.rs (hunk 2 above)
Comment (steveej) [reply policy-engine/src/graph.rs]: I'd prefer `actix_web::test` for the tests, shouldn't be too much overhead

File: policy-engine/src/graph.rs [hunk 3]
```diff
@@ -94,3 +94,50 @@ pub(crate) fn index(req: HttpRequest<AppState>) -> Box<Future<Item = HttpRespons
...
+
+    #[cfg(feature = "test-net-private")]
+    mod private;
```
Comment (lucab) [inline policy-engine/src/graph.rs]: Self note: spurious lines.

**Comment by lucab** [comment]
@steveeJ fixed the minor nit and also caught a copy-paste error. PTAL.

**Review by steveej — APPROVED** [review]
This is a nice testing approach.

/lgtm

**Pull request merged**